### PR TITLE
Typescript Support

### LINKED ISSUE #7: Typescript conversion
How does the team feel about converting this library to Typescript?

The advantage is that compiling from Typescript will automatically produce a .d.ts typings file, which downstream projects can use for type completions. Since this project is referenced from the `screeps-typescript-starter`, it would make sense to provide some good typings for it.

The typings can be kept in sync with @types/screeps as we update the Screeps version in the future.

If distributed in compiled form on NPM, this library will still be consumable from Javascript without typings just as it is now:

```
// test.js
const {ScreepsServer} = require('screeps-server-mockup');
```

```
// test.ts
import {ScreepsServer} from 'screeps-server-mockup';
```

---

I am willing to put up a PR for the conversion, if you guys think it would be desirable for longterm maintainability.

### PR DESCRIPTION
**Pull request by brisberg**
Fix #7 

Typescript Support branch is ready for prime time.

Should be up to date with Master.

You can manually test it by running `npm pack` and inspecting the final bundle. You can even unpack this bundle into another project to see that it works correctly, and supplies types to the IDE.

Let me know if you have any questions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [Unreleased]
 
+### Added
+
+- Typescript Support (TODO: Add link to final pull request typescript -> master)
```
Comment (brisberg) [inline CHANGELOG.md:12]: This line will need to be updated during the next release:

Typescript Support ([#42](https://github.com/screepers/screeps-server-mockup/pull/42))

**Review by pyrodogg — APPROVED** [review]
Thanks for your hard work in porting this to TypeScript. I've spend some time over the past few days reviewing and testing the PR and haven't come up with any issues.

File: CHANGELOG.md (hunk 1 above)
Comment (pyrodogg) [reply CHANGELOG.md:12]: Will do

**Comment by brisberg** [comment]
Awesome, and no problem!

**Pull request merged**